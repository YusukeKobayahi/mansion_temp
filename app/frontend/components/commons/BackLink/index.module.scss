@import "~/styles/variables";
@import "~/styles/utils";

.btn {
  display: block;
  margin: 20px auto 0;
  padding: 8px 20px;
  background-color: #f8f6f0;
  border: 2px solid $color-main;
  border-radius: 40px;
  font-size: 16px;
  color: $color-main;
  cursor: pointer;

  @include screenMd {
    padding: 10px 30px;
    font-size: 14px;
  }
}
