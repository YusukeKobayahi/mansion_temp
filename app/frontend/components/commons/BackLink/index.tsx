import Link from "next/link";
import { buildLibraryPagePath } from "~/lib/utils";
import styles from "./index.module.scss";

type BackLinkProps = {
  name: string;
  page: string;
  id: string;
};

const BackLink: React.FC<BackLinkProps> = ({
  name,
  page,
  id,
}: BackLinkProps) => {
  return (
    <Link href={buildLibraryPagePath(page, id)} passHref>
      <button className={styles.btn}>{name}の中古マンション一覧</button>
    </Link>
  );
};

export default BackLink;
