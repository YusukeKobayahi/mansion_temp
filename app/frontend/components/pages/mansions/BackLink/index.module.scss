@import "~/styles/variables";
@import "~/styles/utils";

.container {
  width: 100%;
  padding: 50px calc(50vw - 50%);
  color: $color-text-main;
  box-sizing: content-box;
  margin-bottom: -50px;
  margin: 0 calc(50% - 50vw) -50px;
  background-color: #f5f5f5;
  text-align: center;

  @include screenMd {
    max-width: 100%;
  }
}

a.btn {
  display: inline-block;
  margin: 0 auto;
  padding: 10px 24px;
  background-color: #f8f6f0;
  border: 2px solid $color-main;
  border-radius: 0px;
  font-size: 16px;
  color: $color-main;
  cursor: pointer;
}
