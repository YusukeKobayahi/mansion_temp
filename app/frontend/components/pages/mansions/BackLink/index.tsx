import Link from "next/link";
import styles from "./index.module.scss";
import { buildLibraryPagePath } from "~/lib/utils";

export interface BackLinkProps {
  jisCode: string;
}
//使用されていないので削除予定
const BackLink: React.FC<BackLinkProps> = ({ jisCode }: BackLinkProps) => (
  <section className={styles.container}>
    <Link href={buildLibraryPagePath("cities", jisCode)} prefetch={false}>
      <a className={styles.btn}>一覧に戻る</a>
    </Link>
  </section>
);

export default BackLink;
