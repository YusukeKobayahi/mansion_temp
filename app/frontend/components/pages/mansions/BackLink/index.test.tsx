import { render } from "@testing-library/react";
import BackLink from "~/components/pages/mansions/BackLink";

describe("components/pages/mansions/BackLink", () => {
  it("正常に表示される", () => {
    const dummyData = {
      jisCode: "13101",
    };
    const { asFragment } = render(<BackLink {...dummyData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
