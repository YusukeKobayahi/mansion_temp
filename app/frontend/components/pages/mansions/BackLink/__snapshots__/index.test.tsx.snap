// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/mansions/BackLink 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <a
      class="btn"
      href="/cities/13101"
    >
      一覧に戻る
    </a>
  </section>
</DocumentFragment>
`;
